src/AllocatorVault.sol

```diff
@@ -81,19 +81,17 @@ contract AllocatorVault {
 
     // --- constructor ---
 
-    constructor(address roles_, address buffer_, address vat_, bytes32 ilk_, address nstJoin_) {
+    constructor(address roles_, address buffer_, bytes32 ilk_, address nstJoin_) {
         roles = RolesLike(roles_);
 
         buffer = buffer_;
-        vat = VatLike(vat_);
         ilk = ilk_;
         nstJoin = NstJoinLike(nstJoin_);
 
-        require(vat_ == nstJoin.vat(), "AllocatorVault/vat-not-match");
-
+        vat = VatLike(nstJoin.vat());
```

Comment:
I'd say to keep more symmetry with the `nst()` call, let's make `nstJoin.vat()` to return directly `VatLike` type.

Reply:
done